app/scripts/models/user.js

```diff
@@ -222,6 +222,22 @@ define([
       return (Session.email && Session.sessionToken &&
         (! Session.cachedCredentials ||
         Session.cachedCredentials.email !== Session.email));
+    },
+
+    signInAccount: function (account, relier) {
```

Comment:
YES! I'll be happy to update the PR I showed you to continue with these changes and rip a lot of this account management logic out of the views.